src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs

```diff
@@ -593,11 +593,9 @@ internal int GetItemsOnLastLine(int section, int itemsPerLine)
 			int remainder = GetDisplayGroupCount(section) % itemsPerLine;
 			return remainder == 0 ? itemsPerLine : remainder;
 		}
-
+		
 		protected virtual void OnItemsSourceChanged(DependencyPropertyChangedEventArgs e)
 		{
-			Items?.Clear();
```

Comment:
We may want to move this to the else of the outermost "if" in `UpdateItems` or clear the items only if the source becomes null.

Reply:
There is no sync between Items and ItemsSource anywhere but here. And this call causes UpdateItems to be called twice (once due to VectorChanged event on Items, once in ItemSourceChanged). 
In any case, GetItems method (which is used to select the correct item source) will use ItemsSource first, and if empty (not an ICollection in fact), it will use Items.